### DISCUSSION TITLE
Website with subtitles generation and audio to text conversion

### DISCUSSION THREAD

**Discussion by Kabanosk** [openai/whisper#147] [Show and tell]
Hi! Today I heard about this super open-source project. I want to contribute by making simple web application, which will be GUI for this model. So I spent couple hours and I would be glad if someone help me improve it. [Repo](https://github.com/Kabanosk/whisper-website) with the website is on my [github profile](https://github.com/Kabanosk). 

Edit#1: You can download subtitles in vtt and srt file type.
Edit#2: I added hosting with Docker.
Edit#3: I added docker compose.

**Comment by mzwoelfer** [comment]


**Reply by Kabanosk** [reply]
Thanks for your appreciation ;)

**Comment by jmichelin** [comment]
ooh looks fun! I will jump in too

**Comment by Adriankn0rr** [comment]
How have you managed to host the whisper model. I‘ve tried AWS lambda layer, but the model seems to large for AWS. Would be very thankful to figure out a solution to host my Python code along the whisper model.

**Reply by Kabanosk** [reply]
I host it on the localhost, so I don't have that problem. I think that it is possible to download the models and run the inference without downloading model every time. Do you think that will help?

**Reply by jonathanjfshaw** [reply]
You can run whisper itself in the browser: https://github.com/ggerganov/whisper.cpp/tree/master/examples/whisper.wasm

**Reply by dankentfield** [reply]
@Adriankn0rr have you tried using a lambda layer for the model?

**Reply by baibhavKumar1** [reply]
But it's in C language how to work with that?

**Comment by jonathanjfshaw** [comment]
I find this huggng face space to be very well done: https://huggingface.co/spaces/aadnk/whisper-webui

**Comment by gsweats** [comment]
Lol, lead FS web dev and DevOps person. Thinking about moving into this from web. Haha, I'll have to set a reminder to follow you and check how that goes, I think once AI has you, you are its no matter what. There's no going back.

**Reply by Kabanosk** [reply]
Thanks a lot for your kind words, @GregSweats :D

**Comment by Cafe-Soluble** [comment]
Worked well ! Thank you :)

**Reply by Kabanosk** [reply]
Thank you very much! :D

**Comment by RosarioDeveloper** [comment]
What the website url?
